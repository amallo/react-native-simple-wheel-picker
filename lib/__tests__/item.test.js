import React from 'react'
import { TouchableOpacity } from 'react-native'
import TestRenderer, { act } from 'react-test-renderer';
import Item from '../item'

it('renders a snapshot', () => {
    const wrapper = TestRenderer.create(<Item />);
    expect(wrapper.toJSON()).toMatchSnapshot();
})

it('renders a touchable item', () => {
    const onPress = jest.fn()
    const wrapper = TestRenderer.create(<Item onPress={onPress} />);
    const container = wrapper.root.findByProps({ testID: 'container-item' })
    container.props.onPress()
    expect(onPress).toBeCalledTimes(1)
})

it('customizes background color', () => {
    const onPress = jest.fn()
    const wrapper = TestRenderer.create(<Item onPress={onPress} backgroundColor={'red'} />);
    const container = wrapper.root.findByProps({ testID: 'container-item' })
    expect(container.props.style.backgroundColor).toBe('red')
})

it('renders a bold text if selected item', () => {
    const onPress = jest.fn()
    const wrapper = TestRenderer.create(<Item onPress={onPress} isSelected={true} backgroundColor={'red'} />);
    const text = wrapper.root.findByProps({ testID: 'text-item' })
    expect(text.props.style).toContainEqual({ fontWeight: 'bold', "color": "#37474F" })
})

it('renders a normal text if unselected item', () => {
    const onPress = jest.fn()
    const wrapper = TestRenderer.create(<Item onPress={onPress} isSelected={false} />);
    const text = wrapper.root.findByProps({ testID: 'text-item' })
    expect(text.props.style).toContainEqual({ fontWeight: 'normal', "color": "#ADADAD" })
})

it('renders a text content', () => {
    const content = "Hello this is a content"
    const wrapper = TestRenderer.create(<Item content={content} />);
    const text = wrapper.root.findByProps({ testID: 'text-item' })
    expect(text.props.children).toBe(content)
})

it('renders with a fixed height', () => {
    const height = 12
    const wrapper = TestRenderer.create(<Item height={height} />);
    const container = wrapper.root.findByProps({ testID: 'container-item' })
    expect(container.props.style.height).toBe(height)
})