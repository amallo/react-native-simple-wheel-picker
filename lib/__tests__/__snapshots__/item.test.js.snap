// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
      "height": undefined,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="container-item"
>
  <Text
    style={
      Array [
        Object {
          "textAlign": "center",
          "textAlignVertical": "center",
        },
        Object {
          "color": "#ADADAD",
          "fontWeight": "normal",
        },
      ]
    }
    testID="text-item"
  />
</View>
`;
